skymap_scanner/utils/load_scan_state.py

```diff
@@ -124,86 +109,71 @@ def load_scan_state(
-                        LOGGER.debug(" -> failed")
-                        orig_packet = None
-                    if orig_packet is None:
-                        raise RuntimeError("Could not read the input GCD file \"{0}\" from any pre-configured location".format(potential_original_GCD_diff_base_filename))
-            else:
-                # try to load the base file from the various possible input directories
-                GCD_diff_base_handle = None
-                for GCD_base_dir in cfg.GCD_BASE_DIRS:
-                    try:
-                        read_url = os.path.join(GCD_base_dir, potential_original_GCD_diff_base_filename)
-                        LOGGER.debug("reading GCD from {0}".format( read_url ))
-                        GCD_diff_base_handle = filestager.GetReadablePath( read_url )
-                        if not os.path.isfile( str(GCD_diff_base_handle) ):
-                            raise RuntimeError("file does not exist (or is not a file)")
-                    except:
-                        LOGGER.debug(" -> failed")
-                        GCD_diff_base_handle=None
-                    if GCD_diff_base_handle is not None:
-                        LOGGER.debug(" -> success")
-                        break
-                
-                if GCD_diff_base_handle is None:
-                    raise RuntimeError("Could not read the input GCD file \"{0}\" from any pre-configured location".format(potential_original_GCD_diff_base_filename))
-
-                orig_packet = load_GCD_frame_packet_from_file( str(GCD_diff_base_handle) )
-            this_packet = load_GCD_frame_packet_from_file(potential_GCD_diff_base_filename)
+            # For the time being, the code will try to find the corresponding GCD in cfg.DEFAULT_GCD_DIR.
+            # It may be possible to access directly the path stored in source_baseline_GCD, but we ignore this possibility out of simplicity.
+            # If the cache has been produced by the v3 scanner then we end up re-building the same path.
+            # Maybe this will be further simplified in the future.
+            source_baseline_GCD_basename = os.path.basename(source_baseline_GCD)
+            source_baseline_GCD_framepacket = None
+            try:
+                read_path = os.path.join(cfg.DEFAULT_GCD_DIR, source_baseline_GCD_basename)
+                LOGGER.debug("load_GCDQp_state => reading source baseline GCD from {0}".format(read_path))
+                source_baseline_GCD_framepacket = load_framepacket_from_file(read_path)
+            except:
+                LOGGER.debug(" -> failed")
+                source_baseline_GCD_framepacket = None
+            if source_baseline_GCD_framepacket is None:
+                raise RuntimeError(f"load_GCDQp_state => Could not read the source GCD file \"{source_baseline_GCD_metadata}\"")
+            
+            cached_baseline_GCD_framepacket = load_framepacket_from_file(cached_baseline_GCD)
             
```

Comment:
It's too bad Githiub doesn't see the commonalities in these lines--I think this section looks fine. I suppose that's why we have tests too